modules/core/src/copy.cpp

```diff
@@ -564,24 +564,292 @@ Mat& Mat::setTo(InputArray _value, InputArray _mask)
 }
 
 
+#if CV_SIMD
+CV_ALWAYS_INLINE v_uint8x16 v_load_mirror_1( const uchar* ptr );
```

Comment:
This functionality could be implemented via `v_lut` intrinsic. While I understand that shuffling should perform essentially faster and like the idea I prefer to retain platform specific functionality as low as possible. IMO it would be better to add new universal intrinsic. Something like `v_reverse`(that could internally fallback to `v_lut` for platforms that unable/not yet managed to implement this functionality).

Reply:
Using v_lut to implement this would significantly slow it down.  I will look into creating a new v_reverse function.